### PR TITLE
Update Line Chart Props

### PR DESCRIPTION
**Pull request by fanny**
#### What is the purpose of this pull request?
as the title says.

#### What problem is this solving?
- Allows hide the chart axis
- Allows different configs for the `xAxis` and `yAxis`
- Allows different interpolations of the chart
- Allows that the axis label would be changed
- Improve the docs, to include charts for show how can we change the atributtes
- Make Line Chart Responsible, now the container, for default, auto-adjust for the size of the screen :tada: 
- Change `schema` to `config` prop

#### How should this be manually tested?
`yarn install` && `yarn start`

#### Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [X] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,13 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- `hide` prop for allow hide a LineChart axis
```
Comment (kevinch) [inline CHANGELOG.md]:
```suggestion
- `hide` prop to allow hiding a LineChart axis
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,13 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+- `type` prop for allow different interpolations of the LineChart
```
Comment (kevinch) [inline CHANGELOG.md]:
```suggestion
- `type` prop to allow different interpolations of the LineChart
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,13 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+- `tick` prop for allow a customized label on LineChart
```
Comment (kevinch) [inline CHANGELOG.md]:
```suggestion
- `tick` prop to allow a customized label on LineChart
```

File: react/components/EXPERIMENTAL_Charts/LineChart/README.md [hunk 4]
```diff
@@ -65,23 +71,89 @@ const schema = {
 
 ```
 
-#### axis
-The axis property is responsible to change visual appearence of the axis in the chart.
+
+#### xAxis or yAxis
+This axis property are responsible to change visual appearence of the axis in the chart.
```
Comment (kevinch) [inline react/components/EXPERIMENTAL_Charts/LineChart/README.md]:
```suggestion
This axis property is responsible to change the visual appearance of the axis in the chart.
```

File: react/components/EXPERIMENTAL_Charts/LineChart/README.md [hunk 5]
```diff
@@ -33,12 +33,12 @@ from the data to the axis.
...
+const customSchema = {container: { height: 200, width: '40%'}};
 <LineChart
   data={sampleData}
   dataKeys={keys}
-  schema={schema}
   xAxisKey='name'
+  schema={customSchema}
 />
 ```
 - Avoid to do grid competes visually with the data.
```
Comment (kevinch) [inline react/components/EXPERIMENTAL_Charts/LineChart/README.md]: Something wrong here in the sentence.

File: react/components/EXPERIMENTAL_Charts/LineChart/README.md [hunk 6]
```diff
@@ -65,23 +71,89 @@ const schema = {
...
 
-#### axis
-The axis property is responsible to change visual appearence of the axis in the chart.
+
+#### xAxis or yAxis
+This axis property are responsible to change visual appearence of the axis in the chart.
 
 - `tickLine`: If set true, axis tick lines will be drawn.
 - `axisLine`: If set true, axis line will be drawn.
+- `hide`: If set true, the axis do not display in the chart..
```
Comment (kevinch) [inline react/components/EXPERIMENTAL_Charts/LineChart/README.md]:
```suggestion
- `hide`: If set true, the axis does not display in the chart.
```

File: react/components/EXPERIMENTAL_Charts/LineChart/README.md [hunk 7]
```diff
@@ -65,23 +71,89 @@ const schema = {
...
+    container: {height: 300, width: '100%'}, 
+    xAxis:{tick: <CustomizedLabel />}
+  }}
+/>
+```
+
+#### grid
+The grid property is responsible to show a grid inside the chart.
+
+- `horizontal`: If set true, horizontal grid lines will be drawn.
```
Comment (kevinch) [inline react/components/EXPERIMENTAL_Charts/LineChart/README.md]:
```suggestion
- `horizontal`: If set to `true`, horizontal grid lines will be drawn.
```

File: react/components/EXPERIMENTAL_Charts/LineChart/README.md [hunk 8]
```diff
@@ -65,23 +71,89 @@ const schema = {
...
+- `vertical`: If set true, vertical grid lines will be drawn.
```
Comment (kevinch) [inline react/components/EXPERIMENTAL_Charts/LineChart/README.md]:
```suggestion
- `vertical`: If set to `true`, vertical grid lines will be drawn.
```

File: react/components/EXPERIMENTAL_Charts/LineChart/README.md [hunk 9]
```diff
@@ -65,23 +71,89 @@ const schema = {
...
+  schema={{ 
+    xAxis:{tick: true},
+    grid: {vertical: true, horizontal: true}
+  }}
+/>
+```
+
 
 #### container
 The container property is responsible to define the size of box that will render the chart.
```
Comment (kevinch) [inline react/components/EXPERIMENTAL_Charts/LineChart/README.md]:
```suggestion
The container property is responsible to define the size of the box that will render the chart.
```

File: react/components/EXPERIMENTAL_Charts/LineChart/README.md [hunk 10]
```diff
@@ -65,23 +71,89 @@ const schema = {
...
+    xAxis: {tick: true}, 
+    grid: {vertical: false, horizontal: false}
+  }}
+/>
+```
+
+
+### Line props
+
+- `type`: The interpolation type of line. For more details, check the [recharts types](http://recharts.org/en-US/api/Line#type)
```
Comment (kevinch) [inline react/components/EXPERIMENTAL_Charts/LineChart/README.md]: We could maybe directly bring the most common ones here.

File: react/components/EXPERIMENTAL_Charts/LineChart/README.md (hunk 5 above)
Comment (fanny) [reply react/components/EXPERIMENTAL_Charts/LineChart/README.md]: the `to do` should be `makes`? :thinking:

File: react/components/EXPERIMENTAL_Charts/LineChart/README.md (hunk 10 above)
Comment (fanny) [reply react/components/EXPERIMENTAL_Charts/LineChart/README.md]: I think in make something like [this](https://formidable.com/open-source/victory/gallery/interpolation)

File: CHANGELOG.md [hunk 11]
```diff
@@ -6,6 +6,14 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
 
 ## [Unreleased]
+### Added
+- `hide` prop to allow hiding a LineChart axis
+- `type` prop to allow different interpolations of the LineChart
+- `tick` prop to allow a customized label on LineChart
+
+### Changed
+- `axis` prop to `xAxis` and `yAxis` props on LineChart schema
+- default `container` prop to have a `width` and `length` dependent of the screen size
```
Comment (angoncal) [inline CHANGELOG.md]: `width` and `length` or the second one should be `height`?

Comment (fanny) [reply CHANGELOG.md]: Opss, you're right :dizzy_face:

File: react/components/EXPERIMENTAL_Charts/LineChart/README.md (hunk 10 above)
Comment (kevinch) [reply react/components/EXPERIMENTAL_Charts/LineChart/README.md]: You mean the select option?

Comment (fanny) [reply react/components/EXPERIMENTAL_Charts/LineChart/README.md]: Yes!!

Comment (kevinch) [reply react/components/EXPERIMENTAL_Charts/LineChart/README.md]: Sounds great!

Comment (fanny) [reply react/components/EXPERIMENTAL_Charts/LineChart/README.md]: Done!! :smile:

File: react/components/EXPERIMENTAL_Charts/LineChart/index.tsx [hunk 12]
```diff
@@ -67,10 +69,13 @@ LineChart.propTypes = {
   xAxisKey: PropTypes.string.isRequired,
   
   /** The formatter function of value in tooltip. If you return an array, the first entry will be the formatted "value", and the second entry will be the formatted "key" */
-  formatter: PropTypes.func,
+  tooltipFormatter: PropTypes.func,
   
   /** The schema prop changes some styles of the chart. This prop should be given as an object.*/
-  schema: PropTypes.object,
+  config: PropTypes.object,
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Charts/LineChart/index.tsx]: Can this object really have any shape? 🤔

Comment (kevinch) [reply react/components/EXPERIMENTAL_Charts/LineChart/index.tsx]: That would be a good one indeed.

File: react/components/EXPERIMENTAL_Charts/commonProps.ts [hunk 13]
```diff
@@ -1,12 +1,22 @@
 export const commonDefaultProps = {
   container:{
-    height: 300,
-    width: '100%'
+    height: '99%',
```
Comment (kevinch) [inline react/components/EXPERIMENTAL_Charts/commonProps.ts]: Why not 100%?

File: react/components/EXPERIMENTAL_Charts/LineChart/README.md [hunk 14]
```diff
@@ -33,55 +33,180 @@ from the data to the axis.
...
 <LineChart
   data={sampleData}
   dataKeys={keys}
-  schema={schema}
   xAxisKey='name'
+  config={customConfig}
 />
 ```
-- Avoid to do grid competes visually with the data.
+- Avoid makes grid competes visually with the data.
```
Comment (kevinch) [inline react/components/EXPERIMENTAL_Charts/LineChart/README.md]:
```suggestion
- Avoid the grid from visually competing with the data.
```

**Pull request merged**